triton-inference-client.spec

```diff
@@ -0,0 +1,220 @@
+### RPM external triton-inference-client 2.11.0
+%define branch main
+%define github_user triton-inference-server
+%define tag_2_11_0 36cd3b3c839288c85b15e4df82cfe8fca3fff21b
+
+Source: git+https://github.com/%{github_user}/client.git?obj=%{branch}/%{tag_2_11_0}&export=%{n}-%{realversion}&output=/%{n}-%{realversion}.tgz
+BuildRequires: cmake git
+Requires: protobuf grpc cuda 
+
+%prep
+
+%setup -n %{n}-%{realversion}
+
+%build
+
+# locations of CMakeLists.txt
+PROJ_DIR=../%{n}-%{realversion}/src/c++
+CML_CPP=${PROJ_DIR}/CMakeLists.txt
+CML_LIB=${PROJ_DIR}/library/CMakeLists.txt
+
+# remove rapidjson dependence
+sed -i '/RapidJSON CONFIG REQUIRED/,+13d;' ${CML_LIB}
+sed -i '/triton-common-json/d' ${CML_LIB}
+# core repo not needed for grpc-client-only install
+sed -i '/FetchContent_MakeAvailable(repo-core)/d' ${CML_CPP}
+# remove attempts to install external libs
+sed -i '\~/../../_deps/repo-third-party-build/~d' ${CML_LIB}
+sed -i '\~/../../third-party/~d' ${CML_LIB}
+# keep typeinfo in .so by removing ldscript from properties
+sed -i '/set_target_properties/,+5d' ${CML_LIB}
+# change flag due to bug in gcc10 https://gcc.gnu.org/bugzilla/show_bug.cgi?id=95148
+if [[ `gcc --version | head -1 | cut -d' ' -f3 | cut -d. -f1,2,3 | tr -d .` -gt 1000 ]] ; then 
+    sed -i -e "s|Werror|Wtype-limits|g" ${CML_LIB}
+fi
+
+# extracted from https://github.com/triton-inference-server/server/blob/v2.11.0/src/core/model_config.h
+cat << 'EOF' > ${PROJ_DIR}/library/model_config.h
```

Comment:
@kpedro88 , instead of generating these files here, can you please add these generated files as `cmsdist/triton-inference-client/model_config.[h,cc].file` and then add following sources
```
Source1:  triton-inference-client/model_config.h
Source2: triton-inference-client/model_config.cc
```
and then use
```
cp %{_sourcedir}/model_config.h  ${PROJ_DIR}/library/
cp %{_sourcedir}/model_config.cc ${PROJ_DIR}/library/
```